src/main/scala/com/sumologic/sumopush/actor/LogProcessor.scala

```diff
@@ -211,7 +203,31 @@ object LogProcessor extends MessageProcessor {
       }.collect { case (k: String, v: String) => HeaderField(k, v) }.toSeq
       case _ => Seq.empty
     }
-    (payload, fields)
+
+    val payloadWithFields = payloadWithOptions match {
+      case (payloadText, m: Map[String, Any]) if moveFieldsIntoPayload =>
```

Comment:
not sure we can assume the payload value is a map here as a precondition to appending the fields. what happens when the payload is a string?